### DISCUSSION TITLE
Compile IM to convert raw files

### DISCUSSION THREAD

**Discussion by Cruaier** [ImageMagick/ImageMagick#5118] [Help]
Hi there,

I am working on a project to convert raw and psd files into jpg files. One requirement is to keep the size of the compiled files as small as possible, therefore I can't just compile everything.
At the current state, I have decent progress made with compiling the following libraries (in this order):
- libxml2 2.9.12
- libde265 1.0.8
- lcms2 2.13.1
- nasm 2.15.05
- libjpeg-turbo 2.1.3
- libpng 1.6.37
- bzip2 1.0.8
- tiff 4.3.0
- libwebp 1.2.2
- openjp2 2.4.0
- libheif 1.12.0
- LibRaw 0.20.2
- ghostscript 9.56.1
- ImageMagick 7.1.0-29

The result let me convert most raw and psd files. Most, but not all. Sometimes I stil get errors, in particular:

`identify: Unsupported file format or not RAW file`

I am positive that I'm missing something, because running ImageMagick on a fresh Fedora 35 installation (installing IM from the official repo, using IM 6.9), it works totally fine with all files I'm throwing at it. Comparing the IM configuration of my compilation and the package installed IM there are obviously many differences in the number of delegates, which I assume are the reason that some files can't be "identified".

Delegates from compiled version: `bzlib heic jng jp2 jpeg lcms png raw tiff webp xml zlib`
Delegates from package installed version: `bzlib cairo djvu fontconfig freetype gslib gvc jbig jng jp2 jpeg lcms lqr ltdl lzma openexr pangocairo png ps raqm raw rsvg tiff webp wmf x xml zlib`

If anyone can point me in the direction of what is missing, I'd be very glad.
Thanks for reading!

**Comment by snibgo** [comment]
>  Sometimes I stil get errors, in particular:
identify: Unsupported file format or not RAW file

What input file gives that error? Does that file give the same error when using  normal "complete" IM installation?

**Reply by Cruaier** [reply]
At the moment I'm unable to find a file that I am allowed to share publicly which causes this error. Nevertheless I can share with you the identify information IM gives me on the package installed version:

```
Image:
  Filename: 11884443_4_3.dng
  Format: DNG (Digital Negative)
  Class: DirectClass
  Geometry: 6000x4000+0+0
  Units: Undefined
  Colorspace: sRGB
  Type: TrueColor
  Base type: Undefined
  Endianness: Undefined
  Depth: 16-bit
  Channel depth:
    red: 16-bit
    green: 16-bit
    blue: 16-bit
  Channel statistics:
    Pixels: 24000000
    Red:
      min: 0  (0)
      max: 65535 (1)
      mean: 25124.9 (0.383381)
      standard deviation: 11728.9 (0.178972)
      kurtosis: 1.35076
      skewness: 0.365774
      entropy: 0.723685
    Green:
      min: 0  (0)
      max: 65535 (1)
      mean: 16827.9 (0.256778)
      standard deviation: 9417.2 (0.143697)
      kurtosis: 6.47361
      skewness: 1.53719
      entropy: 0.721264
    Blue:
      min: 0  (0)
      max: 65535 (1)
      mean: 12750 (0.194553)
      standard deviation: 7754.18 (0.118321)
      kurtosis: 15.6916
      skewness: 2.80727
      entropy: 0.716679
  Image statistics:
    Overall:
      min: 0  (0)
      max: 65535 (1)
      mean: 18234.3 (0.278238)
      standard deviation: 9633.44 (0.146997)
      kurtosis: 2.80338
      skewness: 1.19337
      entropy: 0.720543
  Rendering intent: Perceptual
  Gamma: 0.454545
  Chromaticity:
    red primary: (0.64,0.33)
    green primary: (0.3,0.6)
    blue primary: (0.15,0.06)
    white point: (0.3127,0.329)
  Background color: white
  Border color: srgb(223,223,223)
  Matte color: grey74
  Transparent color: black
  Interlace: None
  Intensity: Undefined
  Compose: Over
  Page geometry: 6000x4000+0+0
  Dispose: Undefined
  Iterations: 0
  Compression: Undefined
  Orientation: Undefined
  Profiles:
    Profile-xmp: 6537 bytes
  Properties:
    date:create: 2022-05-05T06:31:03+00:00
    date:modify: 2022-05-05T06:31:03+00:00
    dng:camera.model.name: ILCE-7M3
    dng:create.date: 2022-05-05T03:03:40+00:00
    dng:exposure.time: 1/3.0
    dng:f.number: �
                   �	
    dng:focal.length: 16.0
    dng:focal.length.in.35mm.format: 16 mm
    dng:iso.setting: 11.0
    dng:lens: 16.0-35.0mm f/4.0-0.0
    dng:lens.f.stops: 0.00
    dng:make: Sony
    dng:max.aperture.at.max.focal: 0.0
    dng:max.aperture.at.min.focal: 35.0 mm
    dng:max.aperture.value: 4.0
    dng:max.focal.length: 35.0 mm
    dng:min.focal.length: 16.0 mm
    dng:software: Adobe Photoshop Lightroom Classic 11.2 (Macintosh)
    dng:wb.rb.levels: 1.867272 2.015871 1.000000 0.000000
    signature: 8bacfa899a399144c90d4e5d49a660bd681ca23cc2185ae3a30874c7924ca75e
  Artifacts:
    filename: 11884443_4_3.dng
    verbose: true
  Tainted: False
  Filesize: 6.4068MiB
  Number pixels: 24000000
  Pixels per second: 19.469MB
  User time: 1.210u
  Elapsed time: 0:02.232
  Version: ImageMagick 6.9.12-44 Q16 x86_64 17145 https://imagemagick.org
```

> Does that file give the same error when using normal "complete" IM installation?

No, on the "complete" installation it works.

**Comment by snibgo** [comment]
I don't know what the problem is. I can only suggest a process of elimination: ensure that your IM build contains all the delegates as the "complete" IM that works. Does that work? If if does, then gradually remove delegates that seem irrelevant.

**Reply by Cruaier** [reply]
that will take a lot of time...to write a script to compile all delegates and their dependencies..I'll try to figure out what seems most reasonable, but it is really difficult because the output is too minimal.
